freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/PasswordService.java

```diff
@@ -41,48 +54,103 @@
     @Inject
     private GrpcUmsClient umsClient;
 
-    public SetPasswordResponse setPassword(String accountId, String userCrn, String password, Set<String> envs) {
-        String userId = getUserIdFromUserCrn(userCrn);
+    @Inject
+    private AsyncTaskExecutor asyncTaskExecutor;
+
+    // TODO replace cache with database tables
+    private Cache<String, SetPasswordResponse> passwordOperationStatusCache = CacheBuilder.newBuilder()
```

Comment:
I know. I'm trying to get the API updated along w/ the cli so that we don't confuse the field folks too much when the cli times out. I'm actively working on backing this with the database instead.